include/deal.II/lac/trilinos_tpetra_precondition.h

```diff
@@ -0,0 +1,1311 @@
+       *
+       */
+      struct AdditionalData
+      {
+        /**
+         * @brief Constructor.
+         *
+         * Set the parameters to be used in the preconditioner.
+         * The defaults yield the l1 Gauss-Seidel preconditioner specified in
+         * the original publication.
+         *
+         * @param omega The damping factor for the relaxation.
+         * @param eta Threshold parameter for diagonal correction.
+         * @param fix_diagonal Fix small diagonal entries for the inversion?
+         * @param min_diagonal Threshold for fix_diagonal.
+         * @param n_sweeps Number of relaxation sweeps per call to vmult or Tvmult.
+         */
+        AdditionalData(const double omega        = 1,
+                       const double eta          = 1.5,
+                       const bool   fix_diagonal = false,
+                       const double min_diagonal = 0,
+                       const int    n_sweeps     = 1);
+
+        /**
+         * @brief Relaxation damping factor.
+         *
+         */
+        double omega;
+
+        /**
+         * @brief Threshold parameter for diagonal correction.
+         *
+         * The l1 correction for a row i will only be added if it is more than
+         * eta times larger than the original diagonal entry.
+         *
+         */
+        double eta;
+        /**
+         * @brief Whether or not to enlarge entries below a threshold
+         *
+         * If the matrix diagonal contains entries close to zero, their
+         * inversion will be close to division by zero. This can be corrected by
+         * setting this value to <tt>true</tt>. This will result in additional
+         * computational work during initialization.
+         */
+        bool fix_diagonal;
+        /**
+         * @brief Threshold below which entries will be fixed.
+         *
+         * If the threshold is zero (default) only entries which are exactly
+         * zero will be replaced will small nonzero values.
+         *
+         */
+        double min_diagonal;
+        /**
+         * @brief Set how often the preconditioner should be applied during vmult() or Tvmult().
+         *
+         */
+        int n_sweeps;
+      };
+
+      /**
+       * @brief Constructor.
+       *
+       */
+      PreconditionL1GaussSeidel();
+
+      /**
+       * @brief Compute the preconditioner based on the given matrix and parameters.
+       *
+       * @param A The matrix to base the preconditioner on.
+       * @param additional_data The set of parameters to tune the preconditioner.
+       */
+      void
+      initialize(const SparseMatrix<Number, MemorySpace> &A,
+                 const AdditionalData &additional_data = AdditionalData());
+    };
+
+    /**
+     * @brief The class for the SOR preconditioner within Ifpack2.
+     *
+     * If the code is executed MPI-parallel the individual processors will be
+     * connected with an Additive Schwarz method.
```

Comment:
I think you use both spellings, could you please unify this to
```suggestion
     * connected with an additive Schwarz method.
```

Reply:
Done